Source/JavaScriptCore/wasm/js/JSToWasm.cpp

```diff
@@ -207,13 +211,301 @@ void marshallJSResult(CCallHelpers& jit, const TypeDefinition& typeDefinition, c
+        // If you change this, make sure to modify WebAssembly.asm:op(js_to_wasm_wrapper_entry)
+        CCallHelpers jit;
+        CCallHelpers::JumpList exceptionChecks;
+        JIT_COMMENT(jit, "jsToWasm interpreted wrapper");
+        thunk.construct(std::make_shared<InternalFunction>());
+        jit.emitFunctionPrologue();
+
+        // saveJSEntrypointInterpreterRegisters
+        jit.subPtr(CCallHelpers::TrustedImmPtr(Wasm::JSEntrypointInterpreterCallee::SpillStackSpaceAligned), CCallHelpers::stackPointerRegister);
+
+#if CPU(ARM64) || CPU(ARM64E) || CPU(X86_64)
+        CCallHelpers::Address memBaseSlot { GPRInfo::callFrameRegister, -2 * static_cast<long>(sizeof(Register)) };
+        CCallHelpers::Address wasmInstanceSlot { GPRInfo::callFrameRegister, -3 * static_cast<long>(sizeof(Register)) };
+        jit.storePair64(GPRInfo::wasmBaseMemoryPointer, GPRInfo::wasmBoundsCheckingSizeRegister, memBaseSlot);
+        jit.store64(GPRInfo::wasmContextInstancePointer, wasmInstanceSlot);
+#else
+        CCallHelpers::Address wasmInstanceSlot { GPRInfo::callFrameRegister, -1 * static_cast<long>(sizeof(Register)) };
+        jit.store32(GPRInfo::wasmContextInstancePointer, wasmInstanceSlot);
+#endif
+
+        {
+            // Callee[cfr]
+            CCallHelpers::Address functionSlot { GPRInfo::callFrameRegister, sizeof(CPURegister) * 2 + sizeof(Register) };
+            jit.loadPtr(functionSlot, GPRInfo::regWS0);
+        }
+
+        {
+            CCallHelpers::Address calleeSlot { GPRInfo::regWS0, WebAssemblyFunction::offsetOfInterpreterCallee() };
+            jit.loadPtr(calleeSlot, GPRInfo::regWS0);
+        }
+
+#if ASSERT_ENABLED
+        {
+            CCallHelpers::Address identSlot { GPRInfo::regWS0, JSEntrypointInterpreterCallee::offsetOfIdent() };
+            jit.load32(identSlot, GPRInfo::regWS1);
+            jit.move(MacroAssembler::TrustedImm32(0xBF), GPRInfo::regWA0);
+            auto ok = jit.branch32(CCallHelpers::Equal, GPRInfo::regWS1, GPRInfo::regWA0);
+            jit.breakpoint();
+            ok.link(&jit);
+        }
+#endif
+
+        // Allocate stack space (no stack check)
+        {
+            CCallHelpers::Address frameSlot { GPRInfo::regWS0, JSEntrypointInterpreterCallee::offsetOfFrameSize() };
+            jit.load32(frameSlot, GPRInfo::regWS1);
+            jit.subPtr(GPRInfo::regWS1, CCallHelpers::stackPointerRegister);
+        }
+
+        // Prepare frame
+        jit.setupArguments<decltype(operationJSToWasmEntryWrapperBuildFrame)>(CCallHelpers::stackPointerRegister, GPRInfo::callFrameRegister);
+        jit.callOperation<OperationPtrTag>(operationJSToWasmEntryWrapperBuildFrame);
+
+        // Instance
+        {
+            CCallHelpers::Address instanceSlot { GPRInfo::callFrameRegister, CallFrameSlot::codeBlock * 8 };
+            jit.loadPtr(instanceSlot, GPRInfo::wasmContextInstancePointer);
+        }
+
+        // Memory
+        {
+#if CPU(ARM64) || CPU(ARM64E) || CPU(X86_64)
```

Comment:
In C++ code (unlike offlineasm, confusingly) `CPU(ARM64)` includes `CPU(ARM64E)` so you don't need that case.